xwiki-platform-core/xwiki-platform-web/xwiki-platform-web-templates/src/main/resources/templates/macros.vm

```diff
@@ -2642,40 +2642,47 @@ Recursive title display detected!##
       <input type="text" id="${escapedId}-filter" class="xwiki-select-filter"
         placeholder="$escapetool.xml($services.localization.render('web.widgets.select.filter.placeholder'))" />
     </p>
-    <div class="xwiki-select-options">
-      <ul>
-        #set ($count = 0)
-        #foreach ($category in $options)
-          <li class="xwiki-select-category">$category.name (<span class="xwiki-select-category-count">$category.options.size()</span>)
-            <ul>
-              #foreach ($option in $category.options)
-                #if ("$!defaultValue" == $$option.value || ($count == 0 && $firstIsDefaultIfDefaultNull && "$!defaultValue" == ''))
-                  #set ($checked = 'checked="checked"')
-                  #set ($class = 'xwiki-select-option-selected')
-                #else
-                  #set ($checked = '')
-                  #set ($class = '')
-                #end
-                #set ($otherData = '')
-                #foreach($data in $option.data.keySet())
-                  #set ($otherData = "$otherData data-${escapetool.xml($data)} = '$escapetool.xml($option.data.get($data))'")
-                #end
-                <li class="xwiki-select-option $class">
-                  <input type="radio" name="$fieldName" value="$escapetool.xml($option.value)" id="${fieldName}_${count}" $checked $otherData/>
-                  <span class="xwiki-select-option-icon">$!services.icon.renderHTML($option.icon)</span>
-                  <div>
-                    <label for="${fieldName}_${count}">$escapetool.xml($option.name)</label>
-                    #if ("$!option.description" != '')
-                      <p class="xHint">$escapetool.xml($option.description)</p>
-                    #end
-                  </div>
-                </li>
-                #set ($count = $count + 1)
-              #end
-            </ul>
+    <span id="${escapedId}-listbox-label" class="sr-only">
```

Comment:
Why is this `sr-only` while in the very similar similar document extra tabs, the label and the description are hidden?

Reply:
Thanks for the question. I'm writing a proposal for HTML/CSS recommendations so that such an inconsistency won't happen anymore.
Note that sr-only only hides visually and hidden also hides semantically. However,  when referenced with aria-labelledby or aria-describedby, the content is still found and read out even if the node is hidden.
I think that using `sr-only` is still preferable, I don't know if all web browser and assistive techs will consider the above feature as a feature or a bug that needs to be fixed.
I'll fix the PR for XWIKI-20844 where I hid document extra tabs labels with 'hidden'.

Reply:
Here is the proposal for code style: https://forum.xwiki.org/t/css-code-style-hiding-elements/12861

Reply:
Here is the commit on the PR for XWIKI-20844: https://github.com/xwiki/xwiki-platform/pull/2165/commits/19b1cf391baa5ea4920bd157ee146f138d3e550c

Reply:
I've just looked at the [official documentation of aria-labelledby](https://w3c.github.io/aria/#aria-labelledby) and found

> If the interface is such that it is not possible to have a visible label on the screen, authors SHOULD use [aria-label](https://w3c.github.io/aria/#aria-label) and SHOULD NOT use [aria-labelledby](https://w3c.github.io/aria/#aria-labelledby).

which to me clearly indicates that in this case we should prefer `aria-label` over `aria-labelledby`. Any reason why you think `aria-labelledby` is preferable here? Also, the fact that `aria-labelledby` and `aria-describedby` can reference hidden content including HTML elements with `hidden` attribute is included in the official [ARIA Authoring Practices Guide (APG)](https://www.w3.org/WAI/ARIA/apg/practices/names-and-descriptions/#naming_with_aria-labelledby) so it would be very strange if this feature was removed one day. On the other hand, using `sr-only` could lead to content being read twice by screen readers.

Reply:
> Any reason why you think aria-labelledby is preferable here?

I preferred this solution because the label and description (or hint :upside_down_face: ) fields are handled the same way. Unfortunately, [aria-description is not a thing](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Attributes/aria-description) until ARIA 1.3 gets recommended (the project is not close to finishing, so probably in at least a couple years). I like this similarity to make code easier to understand (I don't think it's important enough to make a proposal out of this though ), but I won't die on this hill, let me know what you think we should do here.


> Also, the fact that aria-labelledby and aria-describedby can reference hidden content including HTML elements with hidden attribute is included in the official [ARIA Authoring Practices Guide (APG)](https://www.w3.org/WAI/ARIA/apg/practices/names-and-descriptions/#naming_with_aria-labelledby) so it would be very strange if this feature was removed one day.

You got a point. Addressed in https://github.com/xwiki/xwiki-platform/pull/2182/commits/b7aa61e742929b75751607d787824903576bdf25
I updated https://dev.xwiki.org/xwiki/bin/view/Community/CodeStyle/XhtmlCssCodeStyle/ to highlight this case :)